Genuine failure in CI:
```
2026-07-23T20:33:42.2097953Z [1047/1048] Linking build/sitl_periph_PPP/bin/AP_Periph
2026-07-23T20:33:42.6956431Z /usr/bin/ld: lib/libAP_Periph_libs.a(lua_generated_bindings.cpp.8.o): warning: relocation against `_ZN6AP_OSD10_singletonE' in read-only section `.text._ZN6AP_OSD13get_singletonEv[_ZN6AP_OSD13get_singletonEv]'
2026-07-23T20:33:42.6958619Z /usr/bin/ld: lib/libAP_Periph_libs.a(lua_generated_bindings.cpp.8.o): in function `AP_OSD::get_singleton()':
2026-07-23T20:33:42.6959943Z /__w/ardupilot/ardupilot/build/sitl_periph_PPP/../../libraries/AP_OSD/AP_OSD.h:541:(.text._ZN6AP_OSD13get_singletonEv[_ZN6AP_OSD13get_singletonEv]+0x7): undefined reference to `AP_OSD::_singleton'
2026-07-23T20:33:42.6961115Z /usr/bin/ld: lib/libAP_Periph_libs.a(lua_generated_bindings.cpp.8.o): in function `AP_OSD::draw_screen()':
2026-07-23T20:33:42.6962498Z /__w/ardupilot/ardupilot/build/sitl_periph_PPP/../../libraries/AP_OSD/AP_OSD.h:671:(.text._ZN6AP_OSD11draw_screenEv[_ZN6AP_OSD11draw_screenEv]+0xe): undefined reference to `AP_OSD::update_osd()'
2026-07-23T20:33:42.6963813Z /usr/bin/ld: warning: creating DT_TEXTREL in a PIE
2026-07-23T20:33:42.6964411Z collect2: error: ld returned 1 exit status
```

--- a/libraries/AP_Scripting/generator/description/bindings.desc
+++ b/libraries/AP_Scripting/generator/description/bindings.desc
@@ -1165,8 +1165,8 @@ userdata DroneCAN_Handle method subscribe boolean
 userdata DroneCAN_Handle manual check_message DroneCAN_Handle::check_message 0 4
 userdata DroneCAN_Handle manual_operator __gc DroneCAN_Handle::__gc
 
-include AP_OSD/AP_OSD_Backend.h depends defined(OSD_ENABLED) && OSD_ENABLED
-singleton AP_OSD_Backend depends defined(OSD_ENABLED) && OSD_ENABLED
+include AP_OSD/AP_OSD_Backend.h depends defined(OSD_ENABLED) && OSD_ENABLED && !defined(HAL_BUILD_AP_PERIPH)
+singleton AP_OSD_Backend depends defined(OSD_ENABLED) && OSD_ENABLED && !defined(HAL_BUILD_AP_PERIPH)
 singleton AP_OSD_Backend rename osd
 singleton AP_OSD_Backend get AP_OSD::get_singleton() == nullptr ? nullptr : AP_OSD::get_singleton()->scripting_get_backend()
 singleton AP_OSD_Backend semaphore

